src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -329,6 +329,25 @@ public bool DropSubscriptionConnections(long subscriptionId, SubscriptionExcepti
             return true;
         }
 
+        public bool DeleteAndSetException(long subscriptionId, SubscriptionException ex)
+        {
+            if (_subscriptions.TryRemove(subscriptionId, out SubscriptionConnectionsState state) == false)
+                return false;
+
+            foreach (var connection in state.GetConnections())
+            {
+                // this is just to set appropriate exception, the connections will be dropped on state dispose
+                connection.ConnectionException = ex;
+            }
+
+            state.Dispose();
+
+            if (_logger.IsInfoEnabled)
+                _logger.Info($"Subscription with id '{subscriptionId}' and name '{state.SubscriptionName}' was deleted and connections were dropped.", ex);
```

Comment:
"... was deleted ..." - I  think this might look scary to the user. They might think it deleted the subscription on the server

Reply:
this is actually deletes the subscription

Reply:
Oh, sorry. I thought it was on the client side.